tower-http/src/cors.rs

```diff
@@ -0,0 +1,798 @@
+//! Middleware which adds headers for [CORS][mdn].
```

Comment:
It is minor, but would be good to document behavior of middleware (i.e. how it behavers in case of non-`OPTIONS` and `OPTIONS` etc

Reply:
For sure! I wanna do a pass over all the docs when the implementation is correct.